### PR TITLE
[BACKLOG-22244] Fix the CDF components sometimes not refreshing when …

### PR DESCRIPTION
**Pull request by joaormf**
…the component rendering is longer than the refresh period (legacy code)

[BACKLOG-22553] Refresh the component based on a CDA data source property (legacy code)
[BACKLOG-22080] add properties to avoid blocking when refreshing and to clear the component html element (legacy code)

@marcoslarsen @plagoa @afrjorge @dcleao please review (legacy code)

### CODE REVIEW

File: core-js/src/main/javascript/cdf-legacy/Dashboards.RefreshEngine.js [hunk 1]
```diff
@@ -161,12 +166,20 @@ Dashboards.RefreshEngine = function(){// Manages periodic refresh of components
...
+        //call update, which calls processComponent, but only if not updating
+        //if the component is updating we reinsert it into the queue
+        if(!Dashboards.isComponentUpdating(component)) {
+          refreshComponent(component);
+        }else {
+          insertInQueue(component);
+        }
       }
       if(refreshQueue.length > 0){
         activeTimer = setTimeout("Dashboards.refreshEngine.fireRefresh()", refreshQueue[0].nextRefresh - currentTime );//ToDo: cleaner way to call
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf-legacy/Dashboards.RefreshEngine.js]: Isn't an else missing here? See https://github.com/webdetails/cdf/pull/986/files

Comment (joaormf) [reply core-js/src/main/javascript/cdf-legacy/Dashboards.RefreshEngine.js]: I don't think... the if should always be executed.

Comment (afrjorge) [reply core-js/src/main/javascript/cdf-legacy/Dashboards.RefreshEngine.js]: Oh, PR wasn't linked in the story, I added it.

Comment (joaormf) [reply core-js/src/main/javascript/cdf-legacy/Dashboards.RefreshEngine.js]: https://github.com/webdetails/cdf/pull/991/files

**Pull request merged**